### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by BrittanyBlake**
In this milestone, I created the game logic by implementing the following:
- create a new logic folder with two files 
   - calculate.js
   - operate.js
- Implement the calculate.js module
- implement the operate.js module
- import calculate function into the App.js file and i

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  if (operation === '-') {
+    total = num1.minus(num2);
+  } else if (operation === '+') {
+    total = num1.plus(num2);
+  } else if (operation === 'X') {
+    total = num1.times(num2);
+  } else if (operation === '÷') {
+    total = num1.div(num2);
+  } else if (operation === '%') {
+    total = num1.mod(num2);
```
Comment (bellom) [inline src/logic/operate.js]: ### [Optional]
It is a good practice to use `switch case` once our condition is more than 3;
https://medium.com/the-daily-standup/javascript-switch-statement-vs-if-else-if-ee3111793fa0

Comment (bellom) [inline src/logic/operate.js]: When it comes to `%` we only need one operand right?
example: click `10` then `%` = 0.01

example: `num1 = (num1 *= 0.01).toString();`

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  let total = 0;
+
+  if (operation === '-') {
+    total = num1.minus(num2);
+  } else if (operation === '+') {
+    total = num1.plus(num2);
+  } else if (operation === 'X') {
+    total = num1.times(num2);
+  } else if (operation === '÷') {
+    total = num1.div(num2);
```
Comment (bellom) [inline src/logic/operate.js]: Division by zero is not handled properly yet. Let's fix this.
The reason is that when a division by zero happens, we should display an error message like `zero division error`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @BrittanyBlake  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/logic/operate.js (hunk 1 above)
Comment (BrittanyBlake) [reply src/logic/operate.js]: Thanks for the suggestion!

**Comment by BrittanyBlake** [comment]
Thank you for the review. I have implemented all of the changes requested.

File: src/components/app/App.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Display from '../display/Display';
+import ButtonPanel from '../button-panel/ButtonPanel';
+// eslint-disable-next-line
```
Comment (uryelah) [inline src/components/app/App.js:4]: - [x] [OPTIONAL] Do you really need this ignore comment line? Usually it's a good idea to research how to fix the eslint rule that was broken, that improves your code style and makes for a cleaner file.

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
```
Comment (uryelah) [inline src/logic/calculate.js:3]: - [x] [OPTIONAL] This is very optional, but if you wanted to you could also destructure the dataObject at line 3, ie: `({ total, next, operation }, buttonName)` and shorten your file by one line.

File: src/logic/operate.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      break;
+    case '+':
+      total = num1.plus(num2);
+      break;
+    case 'X':
+      total = num1.times(num2);
+      break;
+    case '÷':
+      if (num2 === 0) {
+        total = 'NaN';
```
Comment (uryelah) [inline src/logic/operate.js:20]: - Great way of dealing with the division by Zero issue.

**Review by uryelah — APPROVED** [review]
Hi **Brittany Blake**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with this milestone! You've implemented the logic very well, your code is clean, easy to read, and using ES6 fluently.
The only thing I'd suggest is that you'd add a screenshot of your calculator to your README file, other than that, and a couple of optional tidbits below, you are good to go! 🚀 

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: src/logic/calculate.js (hunk 4 above)
Comment (BrittanyBlake) [reply src/logic/calculate.js:3]: implemented in milestone 5! Thanks 😄

File: src/components/app/App.js (hunk 3 above)
Comment (BrittanyBlake) [reply src/components/app/App.js:4]: also resolved in milestone 5!

**Pull request merged**